### DISCUSSION TITLE
Writing an image to a buffer leading to Invalid access memory

### DISCUSSION THREAD

**Discussion by SnowdingerrDev** [libvips/libvips#4307] [Q&A] [answered]
Hello coders,

I'm quite new here and currently working on a project to learn how to use VIPS with Java bindings such as JNA. I've been trying to save an image to a buffer with a specific encoding, but I can't seem to find a working solution despite many different tweaks.

I've been trying the following solution based on other discussions, but I keep encountering the same error:
```java
Pointer image = LibvipsTest.loadImage(sourceFile.toAbsolutePath().toString());

int saveResult = VipsLibrary.INSTANCE.vips_image_write_to_buffer(image, "." + format, buffer, size, "Q", encode.getQuality() + "");

PointerByReference buffer = new PointerByReference();
PointerByReference size = new PointerByReference();

// and then here copy the buffer to an output stream
Pointer bufferPointer = buffer.getValue();
long bufferSize = size.getValue().getLong(0); // <- always crash here
byte[] outputBytes = bufferPointer.getByteArray(0, (int) bufferSize);
outputStream.write(outputBytes);
```

Does anyone have a solution? I'm using JDK 11 and JNA for bindings.

**Comment by jcupitt** [comment]
Hi @SnowdingerrDev,

I would use:

https://github.com/lopcode/vips-ffm

It's a full libvips binding for Java and should do exactly what you need.

**Reply by SnowdingerrDev** [reply]
Oh, thank you ! but does it work under a jdk11 project ? the repository seems to mention a minimal jdk version of jdk22

**Reply by jcupitt** [reply]
Sorry, I don't know :(

**Reply by SnowdingerrDev** [reply, marked as answer]
Hi @jcupitt ,
It doesn't seem to be a fit for a java11 project, however, I'd be interested in any resources you may have to fix my usage of the vips bindings to be able to save an image to an output stream

edit: I found the solution while trying random things over and over, the solution was to use this to get the size of the buffer instead of what I was using:
```java
long bufferSize = size.getPointer().getLong(0); // <- getPointer() instead of getValue()
```

**Reply by jcupitt** [reply]
I'm glad you found a fix.

There are other java bindings, though I've not used them myself, for example: https://github.com/criteo/JVips

Even if you can't use them directly, you might be able to check then for ideas.
